src/ape/api/providers.py

```diff
@@ -1032,7 +1032,19 @@ def get_receipt(
         except TimeExhausted as err:
             raise TransactionNotFoundError(txn_hash) from err
 
-        txn = dict(self.web3.eth.get_transaction(HexStr(txn_hash)))
+        network_config = self.network.config.get(self.network.name)
+        max_retries = network_config.get("max_receipt_retries")
+        for attempt in range(max_retries):
+            try:
+                txn = dict(self.web3.eth.get_transaction(HexStr(txn_hash)))
```

Comment:
Are you confident this resolves your issue?

It is little odd to me because we already have the receipt at this point but the txn is the one that is missing?
The receipt checking happens above, this part only fetches the txn which is weird that the chain would know about the receipt and not the txn

Reply:
Definitely confident, I can reliably reproduce the issue without the retry loop, and it never happens with the loop. In local development and use in a personal project over the weekend I added a print statement in the loop body and saw it pretty much every time I ran a script, and that print statement in the loop body would *only* print in conditions that previously would have been an uncaught exception

I think the confusion is mine in that I shouldn't have used the word `receipt` here when it means something else

it is weird, but not that weird. this probably would never hit if you're talking to a single node. when going through infura/alchemy/other public RPC's, there are probably a number of nodes behind the same RPC endpoint, and its conceivable that one node may know about a tx before other nodes have that data gossiped to them

Reply:
oh that makes tons of sense, now I can see how you would get to this state